Lib/test/test_typing.py

```diff
@@ -2759,6 +2759,54 @@ def x(self): ...
         with self.assertRaisesRegex(TypeError, only_classes_allowed):
             issubclass(1, BadPG)
 
+    def test_implicit_issubclass_between_two_protocols(self):
+        @runtime_checkable
+        class CallableMembersProto(Protocol):
+            def meth(self): ...
+
+        # All the below protocols should be considered "subclasses"
+        # of CallableMembersProto at runtime,
+        # even though none of them explicitly subclass CallableMembersProto
+
+        class IdenticalProto(Protocol):
+            def meth(self): ...
+
+        class SupersetProto(Protocol):
+            def meth(self): ...
+            def meth2(self): ...
+
+        class NonCallableMembersProto(Protocol):
+            meth: Callable[[], None]
+
+        class NonCallableMembersSupersetProto(Protocol):
+            meth: Callable[[], None]
+            meth2: Callable[[str, int], bool]
+
+        class MixedMembersProto1(Protocol):
+            meth: Callable[[], None]
+            def meth2(self): ...
+
+        class MixedMembersProto2(Protocol):
+            def meth(self): ...
+            meth2: Callable[[str, int], bool]
+
+        for proto in (
+            IdenticalProto, SupersetProto, NonCallableMembersProto,
+            NonCallableMembersSupersetProto, MixedMembersProto1, MixedMembersProto2
+        ):
+            with self.subTest(proto=proto.__name__):
+                self.assertIsSubclass(proto, CallableMembersProto)
+
+        # These two shouldn't be considered subclasses of CallableMembersProto, however,
+        # since they don't have the `meth` protocol member
+
+        class EmptyProtocol(Protocol): ...
+        class UnrelatedProtocol(Protocol):
+            def wut(self): ...
+
+        self.assertNotIsSubclass(EmptyProtocol, CallableMembersProto)
+        self.assertNotIsSubclass(UnrelatedProtocol, CallableMembersProto)
+
```

Comment:
There's a few more cases contemplated in the code that your tests don't cover:

- A class with non-Mapping `__annotations__`
- A class that has the required fields in its `__annotations__`, but that isn't a Protocol

Reply:
Hmmmm. I don't want to add a test for the first bullet, as I think the behaviour is incorrect on py312+:

```pycon
Python 3.13.0a0 (heads/main:3af2dc7588, Jun 15 2023, 19:43:11) [MSC v.1932 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from typing import *
>>> @runtime_checkable
... class CheckMe(Protocol):
...     def meth(self): ...
...
>>> class Strange(Protocol):
...     __annotations__ = ['meth']
...
>>> Strange.__protocol_attrs__
{'meth'}
>>> issubclass(Strange, CheckMe)
False
```

`Strange` has the same protocol members as `CheckMe`, so should probably be considered a subclass of `CheckMe`, but isn't, because it has a non-Mapping as its `__annotations__`.

We can debate whether we (1) want to reject protocols that have non-mapping `__annotations__` elsewhere in typing.py; (2) want to change the logic here so that protocols with non-mapping `__annotations__` are still considered subclasses if their `__protocol_attrs__` are a superset of the other protocol; or (3) decide it's a silly edge case and we should do nothing.

But I think the behaviour _is_ internally inconsistent here, so I don't want to write a test that enforces that behaviour.

Reply:
Makes sense. I think (1) is the right principled answer, but it doesn't feel worth changing anything, so let's just do (3).

Reply:
Your second bullet is covered by this assertion here:

https://github.com/python/cpython/blob/101d5ec7d7fe122fa81a377c8ab8b562d1add9ee/Lib/test/test_typing.py#L3221

But I'll add some more tests here; I think it makes sense to